flutter-for-android.md

```diff
@@ -0,0 +1,1845 @@
+“page” of an app (think Activity), and a Navigator is a widget that manages 
+routes. A Navigator can push and pop routes to help a user move from screen to 
+screen.
+
+Like Android where you can declare your Activities inside the 
+AndroidManifest.xml, in Flutter you can pass in a Map of named routes to the 
+top level MaterialApp instance
+
+<!-- skip -->
+{% prettify dart %}
+ void main() {
+  runApp(new MaterialApp(
+    home: new MyAppHome(), // becomes the route named '/'
+    routes: <String, WidgetBuilder> {
+      '/a': (BuildContext context) => new MyPage(title: 'page A'),
+      '/b': (BuildContext context) => new MyPage(title: 'page B'),
+      '/c': (BuildContext context) => new MyPage(title: 'page C'),
+    },
+  ));
+}
+{% endprettify %}
+Then you can change to this route by getting an hold of the Navigator and 
+calling this named route.
+
+<!-- skip -->
+{% prettify dart %}
+Navigator.of(context).pushNamed('/b');
+{% endprettify %}
+
+The other popular use-case for Intents is to call external components such as a 
+Camera or File picker. For this, you would need to create a native platform 
+integration (or use an existing library)
+
+See [Flutter Plugins] to learn how to build a native platform integration.
+
+
+
+**How do I handle incoming Intents from external applications in Flutter?** 
+
+This can be achieved by registering the intent you want to handle in the 
+AndroidMainfest.xml to launch MainActivity (the default activity created by 
+Flutter)
+
+[code]
+
+Then in MainActivity you can handle the intent and pass it to Flutter 
+
+[code]
+
+Lastly, in Flutter you can receive this via
+
+[code]
+
+
+
+**What is the equivalent of startActivityForResult?**
+
+The Navigator class which handles all Routing in Flutter can be used to get a 
+result back from a route that you have pushed on the stack. This can be done by 
+await'ing on the Future returned by push. For example if you were to start a 
+location route which let the user select their location, you could do 
+
+<!-- skip -->
+{% prettify dart %}
+Map coordinates = await Navigator.of(context).pushNamed('/location');
+{% endprettify %}
+
+then inside your location route once the user has selected their location you 
+can "pop" the stack with the result
+
+<!-- skip -->
+{% prettify dart %}
+Navigator.of(context).pop({"lat":43.821757,"long":-79.226392});
+{% endprettify %}
+
+# Async UI
+
+**What is the equivalent of runOnUiThread in Flutter?**
+
+Because of Flutters reactive architecture, there is no need to worry about if 
+you are running stuff on the UI thread, as long as you make sure you don't 
+block your network calls.
+
+For example, when making long running calls you should always follow the 
+async/await pattern
+
+<!-- skip -->
+{% prettify dart %}
+loadData() async {
+  String dataURL = "https://jsonplaceholder.typicode.com/posts";
+  http.Response response = await http.get(dataURL);
+  setState(() {
+    widgets = JSON.decode(response.body);
```

Comment:
you may want to point out that this would run on the UI thread. All this Dart runs on the main UI thread :)

Reply:
done, rewritten